web-registry/src/components/organisms/MediaForm/MediaForm.tsx

```diff
@@ -0,0 +1,155 @@
+import React from 'react';
+import { Formik, FormikHelpers } from 'formik';
+
+import { requiredMessage } from 'web-components/lib/components/inputs/validation';
+import OnBoardingCard from 'web-components/lib/components/cards/OnBoardingCard';
+import { UrlType, UrlList } from 'web-components/lib/utils/schemaURL';
+
+import {
+  validate,
+  getProjectPageBaseData,
+  getCompactedPath,
+} from '../../../lib/rdf';
+import { ProjectPageFooter } from '../../molecules';
+import { useProjectEditContext } from '../../../pages/ProjectEdit';
+import { MediaFormSimple } from './MediaFormSimple';
+import { MediaFormLegacy } from './MediaFormLegacy';
+
+import type { MediaValuesSimple, MediaErrorsSimple } from './MediaFormSimple';
+import type { MediaErrorsLegacy, MediaValuesLegacy } from './MediaFormLegacy';
+import type { ShaclGraphByUriQuery, Maybe } from '../../../generated/graphql';
+
+export interface MediaBaseValues {
+  'regen:previewPhoto'?: UrlType;
+  'regen:galleryPhotos'?: UrlList;
+  'regen:videoURL'?: UrlType;
+}
+
+type ValueObject = { '@value': string };
+export interface MediaBaseErrors {
+  'regen:previewPhoto'?: ValueObject;
+  'regen:videoURL'?: ValueObject;
+}
+
+export type MediaValues = MediaValuesSimple | MediaValuesLegacy;
+export type MediaErrors = MediaErrorsSimple | MediaErrorsLegacy;
+
+export function isSimpleValues(
+  _values: MediaValues,
+  ccId?: Maybe<string> | string,
+): _values is MediaValuesSimple {
+  return !!ccId;
+}
+
+function isSimpleErrors(
+  _errors: MediaErrors,
+  ccId?: Maybe<string> | string,
+): _errors is MediaErrorsSimple {
+  return !!ccId;
+}
```

Comment:
Should this be extracted to a utils file like `MediaForm.utils`?

Reply:
@flagrede good idea